Next.js standalone output broken (again) when remote cached due to missing PNPM symlinks

### Current Behavior

Symlinks are not restored when cache is used:

❯ `ls -lah dist/apps/my-project/.next/standalone/node_modules/`

```
total 0
drwxr-xr-x   11 iserpa  staff   352B Oct  9 14:33 .
drwxr-xr-x    6 iserpa  staff   192B Oct  9 14:33 ..
drwxr-xr-x  117 iserpa  staff   3.7K Oct  9 14:33 .pnpm
drwxr-xr-x    4 iserpa  staff   128B Oct  9 14:33 babel-plugin-macros
drwxr-xr-x    5 iserpa  staff   160B Oct  9 14:33 caniuse-lite
drwxr-xr-x    7 iserpa  staff   224B Oct  9 14:33 next
drwxr-xr-x    5 iserpa  staff   160B Oct  9 14:33 react
drwxr-xr-x    6 iserpa  staff   192B Oct  9 14:33 react-dom
drwxr-xr-x    5 iserpa  staff   160B Oct  9 14:33 sass
drwxr-xr-x    5 iserpa  staff   160B Oct  9 14:33 uglify-js
drwxr-xr-x    6 iserpa  staff   192B Oct  9 14:33 webpack
```



### Expected Behavior

When cache is used, symlinks should also be restored:

❯ `ls -lah dist/apps/my-project/.next/standalone/node_modules/`

```
total 0
drwxr-xr-x   11 iserpa  staff   352B Oct  9 14:31 .
drwxr-xr-x    6 iserpa  staff   192B Oct  9 14:31 ..
drwxr-xr-x  117 iserpa  staff   3.7K Oct  9 14:31 .pnpm
lrwxr-xr-x    1 iserpa  staff    64B Oct  9 14:31 babel-plugin-macros -> .pnpm/babel-plugin-macros@3.1.0/node_modules/babel-plugin-macros
lrwxr-xr-x    1 iserpa  staff    57B Oct  9 14:31 caniuse-lite -> .pnpm/caniuse-lite@1.0.30001492/node_modules/caniuse-lite
lrwxr-xr-x    1 iserpa  staff    63B Oct  9 14:31 next -> .pnpm/next@13.4.19_pamseawcqpaoyqfggclsvtomea/node_modules/next
lrwxr-xr-x    1 iserpa  staff    37B Oct  9 14:31 react -> .pnpm/react@18.2.0/node_modules/react
lrwxr-xr-x    1 iserpa  staff    58B Oct  9 14:31 react-dom -> .pnpm/react-dom@18.2.0_react@18.2.0/node_modules/react-dom
lrwxr-xr-x    1 iserpa  staff    35B Oct  9 14:31 sass -> .pnpm/sass@1.60.0/node_modules/sass
lrwxr-xr-x    1 iserpa  staff    45B Oct  9 14:31 uglify-js -> .pnpm/uglify-js@3.17.4/node_modules/uglify-js
lrwxr-xr-x    1 iserpa  staff    68B Oct  9 14:31 webpack -> .pnpm/webpack@5.88.2_l5cbnh2nwjrm2cqen44lw7rug4/node_modules/webpack
```



### GitHub Repo

_No response_

### Steps to Reproduce

1. create a nextjs project with standalone output, using pnpm as package manager
1. setup remote cache with `NX_SELF_HOSTED_REMOTE_CACHE_SERVER`
1. `nx build my-project --skip-nx-cache`
1. `rm -rf dist/apps/my-project/`
1. `nx reset`
1. `nx build my-project`
1. check output in `dist/apps/my-project`


### Nx Report

```shell
NX   Report complete - copy this into the issue template

Node           : 22.14.0
OS             : darwin-arm64
Native Target  : aarch64-macos
pnpm           : 10.4.0

nx                     : 21.0.0
@nx/js                 : 21.0.0
@nx/jest               : 21.0.0
@nx/eslint             : 21.0.0
@nx/workspace          : 21.0.0
@nx/cypress            : 21.0.0
@nx/devkit             : 21.0.0
@nx/esbuild            : 21.0.0
@nx/eslint-plugin      : 21.0.0
@nx/module-federation  : 21.0.0
@nx/next               : 21.0.0
@nx/node               : 21.0.0
@nx/plugin             : 21.0.0
@nx/react              : 21.0.0
@nx/rollup             : 21.0.0
@nx/storybook          : 21.0.0
@nx/vite               : 21.0.0
@nx/web                : 21.0.0
@nx/webpack            : 21.0.0
typescript             : 5.7.3
---------------------------------------
Community plugins:
@jscutlery/semver      : 5.3.1
@nx-tools/nx-container : 6.1.0
@nxkit/playwright      : 3.0.2
ngx-deploy-npm         : 8.2.0
---------------------------------------
Local workspace plugins:
         @imc-trading/webgui
         @imc-trading/nx-web
---------------------------------------
Cache Usage: 385.98 MB / 92.64 GB
```

### Failure Logs

```shell
Error: Cannot find module 'styled-jsx/package.json'
```

### Package Manager Version

pnpm 10.4.0

### Operating System

- [x] macOS
- [x] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

Same as #19502, fix is probably also the same

--- a/packages/nx/src/native/cache/http_remote_cache.rs
+++ b/packages/nx/src/native/cache/http_remote_cache.rs
@@ -109,6 +109,7 @@ impl HttpRemoteCache {
         let tar_gz: Vec<u8> = Vec::new();
         let enc = flate2::write::GzEncoder::new(tar_gz, Compression::default());
         let mut archive = Builder::new(enc);
+        archive.follow_symlinks(false);
         trace!("Created tar file for writing");
 
         let cache_path = Path::new(&cache_directory);

